apps/docs/docs/core/custom.md

```diff
@@ -15,20 +15,31 @@ you can pass a `api` attribute to the `initClient` or `initQueryClient`.
 
 **Here's a basic example: **
 ```typescript
+import { contract } from './some-contract'
+import axios, { Method, AxiosError, AxiosResponse, isAxiosError } from 'axios'
 const client = initClient(contract, {
   baseUrl: "http://localhost:3333/api",
   baseHeaders: {
     'Content-Type': 'application/json'
   },
   api: async ({ path, method, headers, body }) => {
     const baseUrl = 'http://localhost:3333/api' //baseUrl is not available as a param, yet
-    const result = await axios.request({
-      method: method as Method,
-      url: `${baseUrl}/${path}`,
-      headers,
-      data: body,
-    })
-    return { status: result.status, body: result.data }
+    try {
+      const result = await axios.request({
+        method: method as Method,
+        url: `${this.baseUrl}/${path}`,
+        headers,
+        data: body,
+      })
+      return { status: result.status, body: result.data }
+    } catch (e: Error | AxiosError | any) {
+      if (isAxiosError(e)) {
+        const error = e as AxiosError
+        const response = error.response as AxiosResponse
+        return { status: response.status, body: response.data }
+      }
+    }
+    return { status: 500, body: {} }
```

Comment:
I think it makes sense to re-throw the error if it is not an axios error

```suggestion
      }
      
      throw e;
    }
```

Reply:
Yup, you're right; I was worried about the return type conflicting, but TS picks it up correctly.